visual_reasoning_nemo_collection/data/clevr.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (C) IBM Corporation 2018
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import json
+from skimage import io
+
+import torch
+from torch.utils.data import Dataset
+from torch.nn.utils.rnn import pad_sequence
+
+from torchvision.transforms import transforms
+
+from nemo.backends.pytorch.nm import DataLayerNM
+from nemo.core.neural_types import ChannelType, NeuralType, LengthsType, LabelsType
+from nemo.utils.decorators import add_port_docs
+from nemo.collections.nlp.data import WordTokenizer
+
+class CLEVRDataset(Dataset):
+
+    def __init__(self, root, split_name: str, transform=None):
+        super().__init__()
+
+        self._root = root
+        self._split_name = split_name
+        self.transform = transform
+
+        # Create vocabulary if it hasn't been done already
+        if not os.path.isfile("data/CLEVR_v1.0/questions/vocab_questions_words.txt") \
+            or not os.path.isfile("data/CLEVR_v1.0/questions/vocab_answers_sentences.txt"):
+
+            print("Building CLEVR vocabulary...")
+
+            with open("data/CLEVR_v1.0/questions/CLEVR_train_questions.json", "r") as f:
+                self._questions_data = json.load(f)["questions"]
+
+            def make_vocab_words(sentences, output_path):
```

Comment:
Hmmm, why do you want to have it in here?

Please note that this is in fact a tokenization - and in tokenizer you are also doing it, but in a slightly different way...